File: backend/InsuranceAPI/Models/Producer.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace InsuranceAPI.Models;

public partial class Producer
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    public string Firstname { get; set; } = null!;

    public string Lastname { get; set; } = null!;

    public DateTime Joined { get; set; }

    [JsonIgnore]
    public virtual ICollection<Insured> Insureds { get; set; } = new List<Insured>();
}

File: src/app/models/Producer.ts
export class Producer{
    public firstname:string;
    public lastname:string;
    public joined:Date;
    public code:number;
    public id?:number;

    constructor(firstname:string,lastname:string,joined:Date,code:number,id?:number){
        this.firstname = firstname;
        this.lastname = lastname;
        this.joined = joined;
        this.code = code;
        this.id = id;
    }
}